unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -776,6 +773,196 @@ public Quaternion TargetRotation
+                                throw new ArgumentException(
+                                    $"fromRoomTypes must include \"Bedroom\" inside of a Bedroom scene: {scene}. You gave: {fromRoomTypes.ToString()}."
+                                );
+                            }
+                            break;
+                        case 4:
+                            if (!chosenRoomTypes.Contains("bathroom")) {
+                                throw new ArgumentException(
+                                    $"fromRoomTypes must include \"Bathroom\" inside of a Bathroom scene: {scene}. You gave: {fromRoomTypes.ToString()}."
+                                );
+                            }
+                            break;
+                    }
+                }
+
+                if (sceneNumber >= 1 && sceneNumber <= 20) {
+                    sceneType = "train";
+                } else if (sceneNumber <= 25) {
+                    sceneType = "val";
+                } else {
+                    sceneType = "test";
+                }
+            } else {
+                // RoboTHOR scene
+                string chars = scene.Substring(startIndex: "FloorPlan_".Length, length: 2);
+                switch (chars) {
+                    case "Tr":
+                        sceneType = "train";
+                        break;
+                    case "Va":
+                        sceneType = "val";
+                        break;
+                    case "Te":
+                        sceneType = "test";
+                        break;
+                    default:
+                        throw new Exception($"Unknown scene name: {scene}. Please open an issue on allenai/ai2thor.");
+                }
+                if (fromRoomTypes != null) {
+                    if (!chosenRoomTypes.Contains("robothor")) {
+                        throw new ArgumentException(
+                            $"fromRoomTypes must include \"RoboTHOR\" inside of a RoboTHOR scene: {scene}. You gave: {fromRoomTypes.ToString()}."
+                        );
+                    }
+                }
+            }
+
+            switch (sceneType) {
+                case "train":
+                    if (useTrainMaterials.GetValueOrDefault(true) == false) {
+                        throw new ArgumentException("Inside of RandomizeMaterials, cannot set useTrainMaterials=false inside of a train scene.");
+                    }
+                    if (!useTrainMaterials.HasValue) {
+                        useTrainMaterials = true;
+                    }
+                    if (!useValMaterials.HasValue) {
+                        useValMaterials = false;
+                    }
+                    if (!useTestMaterials.HasValue) {
+                        useTestMaterials = false;
+                    }
+                    if (!useExternalMaterials.HasValue) {
+                        useExternalMaterials = true;
+                    }
```

Comment:
These statements can be simplified:
```c#
useTrainMaterials = useTrainMaterials.GetValueOrDefault(true);
useValMaterials = useValMaterials.GetValueOrDefault(false);
useTestMaterials = useTestMaterials.GetValueOrDefault(false);
useExternalMaterials = useExternalMaterials.GetValueOrDefault(true);
```
and similarly for those below.

Reply:
Fixed in https://github.com/allenai/ai2thor/pull/656/commits/4fb4558e21393ad3e557d7a9101a677ce8bf58bb